pkg/ebpf/tracer.go

```diff
@@ -644,16 +637,51 @@ func (t *Tracer) populatePortMapping(mp *bpflib.Map) ([]uint16, error) {
 	return closedPortBindings, nil
 }
 
-func (t *Tracer) determineConnectionDirection(conn *ConnectionStats) ConnectionDirection {
-	if conn.Type == UDP {
-		return NONE
+func (t *Tracer) setConnectionDirections(connections []ConnectionStats) {
+	connRAddrMap := make(map[util.Address][]*ConnectionStats, len(connections))
+
+	for _, conn := range connections{
+		if conn.IPTranslation == nil{
+			connRAddrMap[conn.Dest] = append(connRAddrMap[conn.Dest], &conn)
+		} else {
+			connRAddrMap[conn.IPTranslation.ReplDstIP] = append(connRAddrMap[conn.IPTranslation.ReplDstIP], &conn)
+		}
 	}
 
-	if t.portMapping.IsListening(conn.SPort) {
-		return INCOMING
+	for i := 0; i < len(connections); i++ {
+		if connections[i].Direction == LOCAL{
+			continue
+		}
+		hit, ok := connRAddrMap[connections[i].Source]
```

Comment:
This part should also be IPtranslation aware, and use IPTranslation.ReplDstIP if available